pkg/asset/ignition/bootstrap/content/report.go

```diff
@@ -0,0 +1,56 @@
+package content
+
+const (
+	// ReportSystemdContents is a service that reports the bootstrap progress
+	// via a Kubernetes Event.
+	ReportSystemdContents = `
+[Unit]
+Description=Report the completion of the cluster bootstrap process
+# Workaround for https://github.com/systemd/systemd/issues/1312
+Wants=bootkube.service tectonic.service
+After=bootkube.service tectonic.service
+
+[Service]
+# Workaround for https://github.com/systemd/systemd/issues/1312 and https://github.com/opencontainers/runc/pull/1807
+ExecStartPre=/usr/bin/test -f /opt/tectonic/.bootkube.done
+ExecStartPre=/usr/bin/test -f /opt/tectonic/.tectonic.done
+ExecStart=/opt/tectonic/report-progress.sh /opt/tectonic/auth/kubeconfig
+
+Restart=on-failure
+RestartSec=5s
+
+[Install]
+WantedBy=multi-user.target
+`
+
+	// ReportShFileContents is a script for reporting the bootstrap progress.
+	ReportShFileContents = `#!/usr/bin/env bash
+set -e
+
+KUBECONFIG="$1"
+
+echo "Reporting install progress..."
+
+manifest=$(mktemp)
+trap "rm ${manifest}" EXIT
+
+cat <<EOF > "${manifest}"
+apiVersion: v1
+kind: Event
+metadata:
+  name: bootstrap-complete
+  namespace: kube-system
+involvedObject:
+  namespace: kube-system
+message: cluster bootstrapping has completed
+firstTimestamp: $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+lastTimestamp: $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+count: 1
+source:
+  component: bootstrap
+  host: $(hostname)
+EOF
+
+oc --config="$KUBECONFIG" create --filename "${manifest}"
```

Comment:
this might be prone to failures from transient errors.

Reply:
The service will retry automatically.